pkg/pkgrepository/app_deploy.go

```diff
@@ -30,13 +30,111 @@ func (a *App) deploy(tplOutput string) exec.CmdRunResult {
+        #@ def annotations_are_identical(existing_anns, new_anns):
+        #@   carvel_anns = set(["kapp.k14s.io/disable-original",
+        #@                      "kapp.k14s.io/disable-wait",
+        #@                      "packaging.carvel.dev/package-repository-ref",
+        #@                      "kapp.k14s.io/identity",
+        #@                      "kapp.k14s.io/create-strategy"])
+        #@   return filter(existing_anns, carvel_anns) == filter(new_anns, carvel_anns)
+        #@ end
+
+        #@ def labels_are_identical(existing_labels, new_labels):
+        #@   carvel_labels = set(["kapp.k14s.io/app", "kapp.k14s.io/association"])
+        #@   return filter(existing_labels, carvel_labels) == filter(new_labels, carvel_labels)
+        #@ end
+
+        #@ def is_identical(existing, new):
+        #@   if existing.spec.refName != new.spec.refName:
+        #@     return False, "spec.refName"
+        #@   end
+        #@   if existing.spec.version != new.spec.version:
+        #@     return False, "spec.version"
+        #@   end
+        #@   if struct.decode(existing.spec.template) != struct.decode(new.spec.template):
+        #@    return False, "spec.template"
+        #@   end
+        #@   if not labels_are_identical(struct.decode(existing.metadata.labels), struct.decode(new.metadata.labels)):
+        #@     return False, "metadata.labels"
+        #@   end
+        #@   if not annotations_are_identical(struct.decode(existing.metadata.annotations), struct.decode(new.metadata.annotations)):
+        #@     return False, "metadata.annotations"
+        #@   end
+        #@
+        #@   return True, ""
+        #@ end
+
+        #@ identical, reason = is_identical(data.values.existing, data.values.new)
+        #@ if identical:
+        #@overlay/match by=overlay.all
+        ---
+        metadata:
+          #@overlay/match missing_ok=True
+          annotations:
+            #@overlay/match missing_ok=True
+            kapp.k14s.io/noop: ""
+        #@ elif get_rev(data.values.existing.metadata.annotations) > get_rev(data.values.new.metadata.annotations):
+        #@overlay/match by=overlay.all
+        ---
+        metadata:
+          #@overlay/match missing_ok=True
+          annotations:
+            #@overlay/match missing_ok=True
+            kapp.k14s.io/noop: ""
+        #@ elif get_rev(data.values.existing.metadata.annotations) < get_rev(data.values.new.metadata.annotations):
+        #@ print("replacing existing older rev with newer rev")
+        #@ else:
+        #@   msg = "Error: Conflicting Packages with name " + data.values.existing.spec.refName + "; Package is already present but not identical (mismatch in " + reason +")"
+        #@   print(msg)
+        #@   fail(msg)
+        #@ end
+  resourceMatchers:
+  - apiVersionKindMatcher: {apiVersion: data.packaging.carvel.dev/v1alpha1, kind: Package}
```

Comment:
this should also be applying to PackageMetadata (try adding 2 tap repos for example)